File: b/models/Assignment.js
module.exports = (sequelize, DataTypes) => {
  const Assignment = sequelize.define('Assignment', {
    title: { type: DataTypes.STRING, allowNull: false },
    description: { type: DataTypes.TEXT },
    deadline: { type: DataTypes.DATE, allowNull: false },
    priority: { type: DataTypes.ENUM('Low', 'Medium', 'High', 'Critical'), defaultValue: 'Medium' },
    status: { type: DataTypes.ENUM('ACTIVE', 'ARCHIVED', 'DRAFT'), defaultValue: 'ACTIVE' },
  });
  return Assignment;
};

File: nexus-spring-backend/src/main/java/com/nexus/backend/model/Assignment.java
package com.nexus.backend.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "Assignments")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Assignment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false)
    private LocalDateTime deadline;

    private String priority; // ENUM as String: Low, Medium, High, Critical
    private String status;   // ENUM as String: ACTIVE, ARCHIVED, DRAFT

    @ManyToOne
    @JoinColumn(name = "courseId")
    private Course course;
}
